src/instruction.rs

```diff
@@ -36,935 +50,301 @@ enum Operation {
-            },
-            0x3f => Instruction {
-                op: Operation::CMC,
-                size: 1,
-                cycles: 4
-            },
-            0x40 =>Instruction {
-                op: Operation::MOV(Register::B, Register::B),
-                size: 1,
-                cycles: 5
-            },
-            0x41 =>Instruction {
-                op: Operation::MOV(Register::B, Register::C),
-                size: 1,
-                cycles: 5
-            },
-            0x42 =>Instruction {
-                op: Operation::MOV(Register::B, Register::D),
-                size: 1,
-                cycles: 5
-            },
-            0x43 =>Instruction {
-                op: Operation::MOV(Register::B, Register::E),
-                size: 1,
-                cycles: 5
-            },
-            0x44 =>Instruction {
-                op: Operation::MOV(Register::B, Register::H),
-                size: 1,
-                cycles: 5
-            },
-            0x45 =>Instruction {
-                op: Operation::MOV(Register::B, Register::L),
-                size: 1,
-                cycles: 5
-            },
-            0x46 =>Instruction {
-                op: Operation::MOV(Register::B, Register::M),
-                size: 1,
-                cycles: 7
-            },
-            0x47 =>Instruction {
-                op: Operation::MOV(Register::B, Register::A),
-                size: 1,
-                cycles: 5
-            },
-            0x48 =>Instruction {
-                op: Operation::MOV(Register::C, Register::B),
-                size: 1,
-                cycles: 5
-            },
-            0x49 =>Instruction {
-                op: Operation::MOV(Register::C, Register::C),
-                size: 1,
-                cycles: 5
-            },
-            0x4a =>Instruction {
-                op: Operation::MOV(Register::C, Register::D),
-                size: 1,
-                cycles: 5
-            },
-            0x4b =>Instruction {
-                op: Operation::MOV(Register::C, Register::E),
-                size: 1,
-                cycles: 5
-            },
-            0x4c =>Instruction {
-                op: Operation::MOV(Register::C, Register::H),
-                size: 1,
-                cycles: 5
-            },
-            0x4d =>Instruction {
-                op: Operation::MOV(Register::C, Register::L),
-                size: 1,
-                cycles: 5
-            },
-            0x4e =>Instruction {
-                op: Operation::MOV(Register::C, Register::M),
-                size: 1,
-                cycles: 7
-            },
-            0x4f =>Instruction {
-                op: Operation::MOV(Register::C, Register::A),
-                size: 1,
-                cycles: 5
-            },
+            0x00 | 0x10 | 0x20 | 0x30 | 0x08 | 0x18 | 0x28 | 0x38 => Instruction::NOP,
+            0x01 => Instruction::LXI(Register::B, u16::from_le_bytes([bytes[1], bytes[2]])),
```

Comment:
@arnolth Did we not want to keep the read_imm8 or read_imm16 calls here or did you prefer directly calling the function with the bytes and index?

Reply:
We definitely can, I was thinking that we didn't need to call a function that was calling just calling another function.

Reply:
Cool, yeah I mostly did that to reduce the likelihood of a user error of having to remember the ordering of bytes[x]. Either way is ultimately fine though.